### PR TITLE
feature(#174): add omitIndexes

### PR DESCRIPTION
**Pull request by BjornMelgaard**

### CODE REVIEW

**Comment by char0n** [comment]
@BjornMelgaard 

What about this version ?
```javascript
const { contains, curry, addIndex, reject } = require('ramda');


// helpers
const rejectIndexed = addIndex(reject);
const containsIndex = curry((indexes, val, index) => contains(index, indexes));

const omitIndexes = curry((indexes, list) => rejectIndexed(containsIndex(indexes), list));
```

Seems more declarative, easier to read and predicate function creation is hidden inside currying and not exposed to business logic.

**Comment by BjornMelgaard** [comment]
@char0n yes, like it very much

File: src/omitIndexes.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @sig [i] -> [a] -> [a]
+ * @see {@link http://ramdajs.com/docs/#omit|omit}
+ * @param {!Array} indexes The array of indexes to omit from the new array
+ * @param {!Array} list The array to copy from
+ * @return {!Array} The new array with indexes not on it
+ * @example
+ *
+ * RA.omitIndexes([-1, 1, 3], ['a', 'b', 'c', 'd']); //=> ['a', 'c']
+ */
+
```
Comment (char0n) [inline src/omitIndexes.js]: Pls more `rejectIndexed` and `containsIndex` before the jsdoc to that jsdoc is applied directly to `omitIndexes` and not to `rejectIndexed`.

File: test/omitIndexes.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    eq(RA.omitIndexes(indexes)(arr), expected);
+  });
+
+  it('tests non existing indexes are omitted', function() {
+    const indexes = [-1, 1, 3];
+    const arr = ['a', 'b', 'c', 'd'];
+    const expected = ['a', 'c'];
+
+    eq(RA.omitIndexes(indexes, arr), expected);
+  });
```
Comment (char0n) [inline test/omitIndexes.js]: Pls add tests for 3 more usecases:
 - empty indexes (will return unmodified list)
 - empty list (will return empty list)
 - combination of empty list and empty indexes

Just to be sure we have corner cases covered. I will adjust `pickIndexes` in https://github.com/char0n/ramda-adjunct/issues/181 to be consistent with your `omitIndexes`.

Then we can merge.

**Comment by BjornMelgaard** [comment]
@char0n done

File: src/index.d.ts [hunk 3]
```diff
@@ -551,6 +551,12 @@ declare namespace RamdaAdjunct {
          */
         sliceTo<T>(toIndex: number, list: string|Array<T>): string|Array<T>;
 
+        /**
+         * Returns a partial copy of an array omitting the indexes specified.
+         */
+        omitIndexes<T>(indexes: number[], list: Array<T>): Array<T>;
+        omitIndexes(indexes: number): <T>(list: Array<T>) => Array<T>;
```
Comment (char0n) [inline src/index.d.ts]: should this be `indexes: number[]` ?

File: src/omitIndexes.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+// helpers
+const rejectIndexed = addIndex(reject);
+const containsIndex = curry((indexes, val, index) => contains(index, indexes));
+
+/**
+ * Returns a partial copy of an array omitting the indexes specified.
+ *
+ * @func omitIndexes
+ * @memberOf RA
+ * @since {@link https://char0n.github.io/ramda-adjunct/1.18.1|v1.18.1}
```
Comment (char0n) [inline src/omitIndexes.js]: We're at 1.19.0. You can always find the next version inside package.json.

**Comment by BjornMelgaard** [comment]
@char0n done

**Comment by BjornMelgaard** [comment]
@char0n thanks)
